synopsys-task/src/synopsys-task/tools-parameter.ts

```diff
@@ -501,7 +559,12 @@ export class SynopsysToolsParameter {
 
   private async getAzureRepoInfo(): Promise<AzureData | undefined> {
     let azureOrganization = "";
-    const azureToken = AZURE_TOKEN;
+    const azureToken =
+      SCAN_TYPE === constants.BLACKDUCK_KEY
+        ? BLACKDUCK_AZURE_TOKEN
+        : SCAN_TYPE === constants.COVERITY_KEY
+        ? COVERITY_AZURE_TOKEN
+        : POLARIS_AZURE_TOKEN;
```

Comment:
These can be extracted to a method ex. `getAzureToken(scanType)` and you can use that method here.